lib/spack/spack/cmd/buildcache.py

```diff
@@ -495,6 +498,14 @@ def install_tarball(spec, args):
     else:
         tarball = bindist.download_tarball(spec)
         if tarball:
+            if args.sha256:
+                checker = spack.util.crypto.Checker(args.sha256)
+                msg = ('cannot verify checksum for "{0}"'
+                       ' [expected={1}]')
+                msg = msg.format(tarball, args.sha256)
+                assert checker.check(tarball), msg
```

Comment:
We should probably use an explicit exception instead of an `assert` here, since `assert` statements can be turned off with your python options (and should only be used for double-checking things that should always be true, rather than error checking on inputs).

Reply:
Fixed in ef700bd